src/test/java/edu/cornell/kfs/concur/batch/service/impl/MockConcurRequestV4Backend.java

```diff
@@ -0,0 +1,204 @@
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4StatusDTO;
+
+public class MockConcurRequestV4Backend {
+
+    private static final int MAX_RESULT_LIMIT = 100;
+
+    private static final Set<String> ALLOWED_REQUEST_LIST_QUERY_PARAMETERS = Set.of(
+            ConcurApiParameters.VIEW, ConcurApiParameters.START, ConcurApiParameters.LIMIT,
+            ConcurApiParameters.MODIFIED_AFTER, ConcurApiParameters.MODIFIED_BEFORE,
+            ConcurApiParameters.SORT_FIELD, ConcurApiParameters.SORT_ORDER);
+
+    private ConcurrentMap<String, RequestEntry> travelRequests;
+    private String baseRequestV4Url;
+
+    public MockConcurRequestV4Backend(String baseRequestV4Url, RequestV4DetailFixture... requestsToAdd) {
+        this.travelRequests = new ConcurrentHashMap<>();
+        this.baseRequestV4Url = baseRequestV4Url;
+        addTravelRequests(requestsToAdd);
+    }
+
+    public void addTravelRequests(RequestV4DetailFixture... requestsToAdd) {
+        for (RequestV4DetailFixture travelRequest : requestsToAdd) {
+            travelRequests.put(travelRequest.id, new RequestEntry(travelRequest, baseRequestV4Url));
+        }
+    }
+
+    public Optional<ConcurRequestV4ReportDTO> findRequest(String id) {
+        RequestEntry matchingEntry = travelRequests.get(id);
+        return Optional.ofNullable(matchingEntry)
+                .map(entry -> entry.requestDetail);
+    }
+
+    public ConcurRequestV4ListingDTO findRequests(Map<String, String> queryParameters) {
+        Set<String> unsupportedParameters = searchForUnsupportedRequestListQueryParameters(queryParameters);
+        if (!unsupportedParameters.isEmpty()) {
+            throw new IllegalArgumentException("Query contains unsupported parameters "
+                    + "(and this mock backend might not support all the parameters from the real Concur web service): "
+                    + unsupportedParameters.toString());
+        }
+        
+        String view = getExistingParameter(queryParameters, ConcurApiParameters.VIEW);
+        int startIndex = getExistingIntParameter(queryParameters, ConcurApiParameters.START);
+        int limit = getExistingIntParameter(queryParameters, ConcurApiParameters.LIMIT);
+        Date modifiedAfter = getExistingDateParameter(queryParameters, ConcurApiParameters.MODIFIED_AFTER);
+        Date modifiedBefore = getExistingDateParameter(queryParameters, ConcurApiParameters.MODIFIED_BEFORE);
+        String sortField = getExistingParameter(queryParameters, ConcurApiParameters.SORT_FIELD);
+        String sortOrder = getExistingParameter(queryParameters, ConcurApiParameters.SORT_ORDER);
+        
+        Comparator<ConcurRequestV4ListItemDTO> requestSorter = Comparator.comparing(
+                ConcurRequestV4ListItemDTO::getStartDate, Comparator.reverseOrder());
+        
+        if (!StringUtils.equalsIgnoreCase(RequestV4Views.APPROVED, view)) {
+            throw new IllegalArgumentException(
```

Comment:
Is this error handling needed?  Since this backend  is just for testing, I am wondering if it could be simplified.

Reply:
Some of the parameters with dynamic values could end up causing unexpected results or failures if the wrong values are passed in. For some of the parameters that are always expected to have the same values (like the "view" parameter), I could remove the error checks or reduce the exceptions to warnings.